src/lib.rs

```diff
@@ -186,6 +182,32 @@ impl FromIterator<AaConfig> for AaSupport {
     }
 }
 
+#[cfg(feature = "wgpu")]
+#[derive(Error, Debug)]
+pub enum RendererError {
+    #[cfg(feature = "wgpu-profiler")]
+    #[error("couldn't create wgpu profiler")]
+    ProfilerCreationError(#[from] wgpu_profiler::CreationError),
+}
+
+#[cfg(feature = "wgpu")]
+#[derive(Error, Debug)]
+pub enum RenderError {
+    #[error("error while run recording")]
+    WgpuRecordingError(#[from] WgpuRecordingError),
+}
+
+#[cfg(feature = "wgpu")]
+#[derive(Error, Debug)]
+pub enum AsyncRenderError {
+    #[error("error while run recording")]
+    WgpuRecordingError(#[from] WgpuRecordingError),
+    #[error(transparent)]
+    BufferAsyncError(#[from] wgpu::BufferAsyncError),
+    #[error("channel was closed")]
+    ChannelWasClosed,
+}
+
```

Comment:
I don't like this as it is... I'd be a fan of some top-level error type like `VelloError`, this feels a little error-heavy.

In addition, when I read this, I think "in what context would a channel be closed? I'm trying to render." - So the errors aren't helpful. Every error should be an error that a benefits a developer to identify a mistake or change their control flow.

Reply:
A root-level error is common for targeted crates. I could be persuaded if there's a reason we shouldn't use a root-level error type. But in my opinion, Vello is doing 1 job, so I don't see why we need multiple error types everywhere.

The error types I would expect to see, which matter to a developer:
```
enum VelloError {
    // The target does not support WebGPU, e.g. your web browser (wasm32)
    WebGpuNotSupported,
    // Features that WebGpu needs are not available (e.g. Compute shaders)
    GpuFeaturesNotAvailable(list),
    // Failed to get or create a surface for the GPU device
    SurfaceCreation(#[from] wgpu::Error)
    // Shader errors, wgpu pipeline errors, etc.
    Render(#[from] source)
}
```